metallb/base/kustomization.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: metallb
+
+resources:
+  - https://raw.githubusercontent.com/metallb/metallb/v0.9.5/manifests/namespace.yaml
+  - https://raw.githubusercontent.com/metallb/metallb/v0.9.5/manifests/metallb.yaml
```

Comment:
This manifest is deploying `ClusterRole`s `ClusterRoleBinding`s: These resources can't be part of the app and have to go to the `cluster-scope`.

Not sure what privileges are required to deploy `PodSecurityPolicy` though.. This probably also require cluster admin, right?

Reply:
So there's really no way to consume an upstream resource directly with our current repository structure, I guess? We have to pull the manifests locally and split things up by hand.

Reply:
If they mix cluster scoped and namespaced resources, nope, there's no way to do that - because you don't want any remote references for the cluster scoped ones - remote can change and we don't want to automatically pull in different manifests and apply them with cluster admin permissions. It's neither deterministic nor secure. 

If you can get the cluster scoped/namespaced manifests as separate remote manifests, you can use the namespaced as a remote reference, there's no big risk in that, since you can only compromise your own namespace.

The cluster scoped resources have to be parsed out and adopted to our directory structure in `cluster-scope` any way.

Reply:
Is the fact that this splits the serviceaccount (which is namespaced) and the clusterrolebinding that references it going to cause any problems?

Reply:
It doesn't. Because of additive nature of OpenShift RBAC. `ClusterRoleBinding` is allowing something to somebody. Therefore the user/sa has to identify first in order for crb to be used - and if there's no sa to identify it just dosn't get used... That's why you can really put anything in the subject spec and it doesn't cause troubles until the user identifies. :smile: It's weird sometimes, but handy in other times. :slightly_smiling_face: